m2cgen/ast.py

```diff
@@ -7,6 +7,10 @@
 
 class Expr:
     output_size = 1
+
+    # The precedence of this expression. Higher value means higher precedence.
+    precedence = None
```

Comment:
Ah, I'm somewhat unsure about this one. On one hand I agree that this simplifies the code quite a bit. On the other hand, I'm hesitant to introduce `precedence` for expressions to which this concept doesn't even apply. `None` in this case means "not applicable", while 0 represents an actual precedence. I feel like I'd like to preserve this distinction and keep it explicit to avoid unexpected behavior / unmet expectations in future. What do you think?

Reply:
> I'm hesitant to introduce `precedence` for expressions to which this concept doesn't even apply.

Makes sense! After your clarification I'd like to preserve `None` as default value too 😃